src/workloads/query/ShardFilter.yml

```diff
@@ -0,0 +1,101 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests the performance of queries which need to perform shard filtering.
+Keywords:
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  Namespace: &Namespace test.Collection0
+  MaxPhases: &MaxPhases 8
+  DocumentCount: &DocumentCount 1000
+
+Actors:
+- Name: CreateShardedCollection
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: admin
+        Operations:
+        - OperationMetricsName: EnableSharding
+          OperationName: AdminCommand
+          OperationCommand:
+            enableSharding: *Database
+        - OperationMetricsName: ShardCollection
+          OperationName: AdminCommand
+          OperationCommand:
+            shardCollection: *Namespace
+            key: {sKey: 1}
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Collection: *Collection
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: *DocumentCount
+        Document:
+          sKey: {^RandomInt: {min: 0, max: 10000}}
+          a: {^RandomInt: {min: 0, max: 10000}}
+
+- Name: QueryByShardKey
+  Type: CrudActor
+  Database: *Database
+  Threads: 16
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {sKey: {$gt: 5000}}
```

Comment:
Are we making an intentional choice here about the cardinality of this query (50% of the collection)? If so, we should document it in a comment.

Reply:
Mostly I just want to see multiple cardinalities of queries being tested, unless we're trying to force a particular scenario to occur. I think the extra tests you added should cover that.

Dumb question- do we still shard-filter if we match no documents? If so, maybe a no-match case would be helpful.